hazelcast/src/main/java/com/hazelcast/monitor/TimedMemberState.java

```diff
@@ -155,6 +169,9 @@ public void fromJson(JsonObject json) {
         memberState = new MemberStateImpl();
         memberState.fromJson(jsonMemberState);
         sslEnabled = getBoolean(json, "sslEnabled", false);
+        if (memberState.getNodeState().getClusterVersion().isGreaterOrEqual(Versions.V3_9)) {
```

Comment:
Members send messages to each other based on the cluster version, even if their codebase version is newer. This way, they can continue working together during a rolling upgrade. I think it should be the same with the Management Center related data that they send. Vassilis said something similar in the independent mancenter release plugin PRD.

However, on second though, I'm not sure if it's really required in the ManCenter context though. We will already require the users to upgrade their ManCenter version before a rolling upgrade. This will be required only when they don't upgrade ManCenter before starting out the rolling upgrade process.

@vbekiaris What do you think? Do you think there's any good reason to do this (conditional serialization based on cluster version), keeping in mind that we recommend the user to upgrade ManCenter version before starting out with rolling upgrade process?